### PR TITLE
Document handling of glibc-compat libraries. NFC

### PR DESCRIPTION
**Pull request by sbc100**
.

### CODE REVIEW

**Comment by sbc100** [comment]
Updated to be a pure NFC

File: tools/building.py [hunk 1]
```diff
@@ -1351,15 +1349,27 @@ def map_to_js_libs(library_name):
...
     'openal': ['library_openal.js'],
-    'rt': [],
-    'pthread': [],
     'X11': ['library_xlib.js'],
     'SDL': ['library_sdl.js'],
-    'stdc++': [],
     'uuid': ['library_uuid.js'],
-    'websocket': ['library_websocket.js']
+    'websocket': ['library_websocket.js'],
+    # These 4 libraries are seperate undef glibc but are all rolled into
```
Comment (kripken) [inline tools/building.py:1357]:
```suggestion
    # These 4 libraries are seperate under glibc but are all rolled into
```

File: tools/building.py [hunk 2]
```diff
@@ -1351,15 +1349,27 @@ def map_to_js_libs(library_name):
...
+    # libc with musl.  For compatability with glibc we just ignore them
```
Comment (kripken) [inline tools/building.py:1358]:
```suggestion
    # libc with musl.  For compatibility with glibc we just ignore them
```

File: tools/building.py [hunk 3]
```diff
@@ -1351,15 +1349,27 @@ def map_to_js_libs(library_name):
...
+    'm': [],
+    'rt': [],
+    'pthread': [],
+    # This is the name of GNU's C++ standard library. We ignore it here
+    # for compatability with GNU toolchains.
+    'stdc++': [],
+    # This means that linking against libc with `-lc` is ignored when the
+    # library is not found (It does work if library already exists in the
+    # sysroot because that case is handled before we call (map_to_js_libs).
+    # TODO(sbc): We should probably remove this and allow `-lc` to make to all
```
Comment (kripken) [inline tools/building.py:1370]:
```suggestion
    # TODO(sbc): We should probably remove this and allow `-lc` to make it all
```

**Review by kripken — APPROVED** [review]

**Pull request merged**